membership-attribute-service/app/services/IdentityAuthService.scala

```diff
@@ -1,24 +1,38 @@
 package services
 
 import _root_.play.api.mvc.RequestHeader
-import com.gu.identity.play
-import com.gu.identity.play.{AccessCredentials, AuthenticatedIdUser}
-import configuration.Config
-
-object IdentityAuthService extends AuthenticationService {
-  val playAuthService = new play.AuthenticationService {
-    override val identityKeys = Config.idKeys
-
-    override lazy val authenticatedIdUserProvider =
-      AuthenticatedIdUser.provider(
-        AccessCredentials.Cookies.authProvider(identityKeys),
-        AccessCredentials.Token.authProvider(identityKeys,"membership")
-      )
+import com.gu.identity.{IdapiConfig, play}
+import org.http4s.Uri
+import cats.implicits._
+import com.gu.identity.model.User
+import com.gu.identity.play.IdentityPlayAuthService
+import com.gu.identity.play.IdentityPlayAuthService.UserCredentialsMissingError
+import com.gu.monitoring.SafeLogger
+import com.gu.monitoring.SafeLogger._
+import scala.concurrent.{ExecutionContext, Future}
+
+class IdentityAuthService(apiConfig: IdapiConfig)(implicit ec: ExecutionContext) extends AuthenticationService {
+
+  val idApiUrl = Uri.unsafeFromString(apiConfig.url)
+
+  val identityPlayAuthService = IdentityPlayAuthService.unsafeInit(idApiUrl, apiConfig.token, Some("membership"))
+
+  def user(implicit requestHeader: RequestHeader): Future[Option[User]] = {
+    getUser(requestHeader)
+      .map(user => Option(user))
+      .handleError { err =>
+        if(err.isInstanceOf[UserCredentialsMissingError])
+          SafeLogger.error(scrub"unable to authorize user - $err", err)
```

Comment:
`identity-auth-play` will try to authenticate the user and this needs either a `SC_GU_U` cookie or a `GU-IdentityToken` header, and will throw a `UserCredentialsMissingError` the request doesn't have appropriate credentials to even try and authenticate with IDAPI. 

It seems like UserCredentialsMissingError should be an error log (something is submitting bad requests to members-data-api) 

IDAPI returning unauthorised can actually just be business as usual, because the cookie/token is expired or otherwise invalid, so I'm suggesting putting that as a warn.

Reply:
I got these backwards in https://github.com/guardian/members-data-api/pull/393, so the log levels in this handleError block and the version of identity-auth-play are the only changes in this PR vs https://github.com/guardian/members-data-api/pull/393.

Reply:
might be worth making it clearer in the warning message that it's an expected case, just in case someone sees it in future

Reply:
```suggestion
          SafeLogger.error(scrub"invalid request as no token provided - unable to authorize user - $err", err)
```